### PR TITLE
Allow httpd password to have dollar sign in it

### PR DESCRIPTION
**Pull request by yrudman**
Having `$` sign inside double quotation for echo argument is interpreted as variable which need to be resolved.

https://bugzilla.redhat.com/show_bug.cgi?id=1438974

@miq-bot add-label bug

\cc @bdunne

### CODE REVIEW

**Comment by yrudman** [comment]
@miq-bot add-label fine/yes

File: lib/gems/pending/appliance_console/external_httpd_authentication.rb [hunk 1]
```diff
@@ -128,7 +128,7 @@ def configure_sssd
 
     def configure_ipa_http_service
       say("Configuring IPA HTTP Service and Keytab ...")
-      AwesomeSpawn.run!("/bin/echo \"#{@password}\" | /usr/bin/kinit #{@principal}")
+      AwesomeSpawn.run!("/bin/echo '#{@password}' | /usr/bin/kinit #{@principal}")
```
Comment (bdunne) [inline lib/gems/pending/appliance_console/external_httpd_authentication.rb]: @yrudman Is it possible to add some tests for this?

**Comment by Fryguy** [comment]
Then you can't have `"` in your password either. The problem here is that awesome_spawn is being used incorrectly...This should be

```ruby
AwesomeSpawn.run!("/usr/bin/kinit", :params => [@principal], :stdin_data => @password)
```

**Comment by yrudman** [comment]
thank you @Fryguy

File: spec/appliance_console/external_httpd_authentication_spec.rb [hunk 2]
```diff
@@ -162,4 +163,26 @@
       subject.post_activation
     end
   end
+
+  context "#configure_ipa_http_service" do
+    before do
```
Comment (bdunne) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: Why use a `before` block when there is only one test?

File: spec/appliance_console/external_httpd_authentication_spec.rb [hunk 3]
```diff
@@ -162,4 +163,26 @@
...
+      allow(subject).to receive(:say)
```
Comment (bdunne) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: Shouldn't we `expect` all of these things to happen?

File: spec/appliance_console/external_httpd_authentication_spec.rb [hunk 4]
```diff
@@ -162,4 +163,26 @@
...
+      service = double("Principal")
+      allow(ApplianceConsole::Principal).to receive(:new).and_return(service)
+      allow(service).to receive(:register)
+      allow(service).to receive(:name)
+      allow(service).to receive(:name)
```
Comment (bdunne) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: Prefer `expect(service).to receive(:name).twice`

File: spec/appliance_console/external_httpd_authentication_spec.rb (hunk 2 above)
Comment (yrudman) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: I think it looks more clean to separate what we are testing from what is not important

File: spec/appliance_console/external_httpd_authentication_spec.rb (hunk 4 above)
Comment (yrudman) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: my mistake, forgot to delete that line, we need that only once

File: spec/appliance_console/external_httpd_authentication_spec.rb (hunk 3 above)
Comment (yrudman) [inline spec/appliance_console/external_httpd_authentication_spec.rb]: I was thinking that `before` block is setting-up environment for executing expectation in `it' block. I am not sure what is the right way (`expect` vs `allow`) to mock some behavior  which is not tested in `it`. 
@bdunne, do you think it make sense to remove `before` and replace all `allow` with `expect` here?

**Comment by simaishi** [comment]
Fine backport details:

```
$ git log -1
commit 5be8085f865e889b33b2f6871318dcf97b0f9305
Author: Brandon Dunne <brandondunne@hotmail.com>
Date:   Tue Aug 15 11:57:23 2017 -0400

    Merge pull request #257 from yrudman/allow-httpd-password-to-have-dollar-sign
    
    Allow httpd password to have dollar sign in it
    (cherry picked from commit 7bea4f3d0ebda0ef479110b7c61e42e9dbf72834)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1481846
```

**Pull request merged**